### PR TITLE
Create User model

### PR DESCRIPTION
**Pull request by joyapisi**
## Work Done:

- Create four models: User, Post, Comment and Like
- Add additional methods to the models:

  - Users: 
    - A method that returns the 3 most recent posts for a given user.
  - Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
  - Comments
    - A method that updates the comments counter for a post.
  - Likes
    - A method that updates the likes counter for a post.
    - Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  has_many :posts
```
Comment (mikethreels) [inline app/models/user.rb]: - [ ] Please ensure to include setting the foreign key for example `has_many :posts, foreign_key: 'author_id'`
- [ ] in the User class the has_many side of the one-to-many relationship between user and like and user and comment should be set as well, make sure to include the foreign as mentioned for the posts.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
```
Comment (mikethreels) [inline app/models/post.rb:3]: - [ ]  in the Post class the has_many side of the one-to-many relationship between posts and like should be set as well.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author
```
Comment (mikethreels) [inline app/models/like.rb]: - [ ] Please make sure to refer to the class in order for this association to work exactly like you did in the Post class.

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (mikethreels) [inline app/models/comment.rb]: - [x] no need to mention the foreign_key in this part of the association, this should be mentioned the has_many side.

File: app/models/like.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  belongs_to :post
+
+  def update_like_posts_counter
+    post.update(likes_counter: post.likess.count)
```
Comment (mikethreels) [inline app/models/like.rb]: - [ ] it seems you might have made a typo in this line because you typed 'likess' with double "ss"

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Joy, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job creating the recent_posts methods.

## Required Changes ♻️
- [ ] the link to the MIT license in the readme file is not working as expected, please have a look at the Microverse template in case you're unsure how it should work.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @joyapisi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**